lightning-net-tokio/src/lib.rs

```diff
@@ -229,7 +230,8 @@ impl Connection {
 		// we shove a value into the channel which comes after we've reset the read_paused bool to
 		// false.
 		let (read_waker, read_receiver) = mpsc::channel(1);
-		let (reader, writer) = io::split(stream);
+		stream.set_nonblocking(true).unwrap();
+		let (reader, writer) = io::split(TcpStream::from_std(stream).unwrap());
```

Comment:
I'd initially gone with `tokio::net::TcpStream` in the public API but then realized its much neater to do the `set_nonblocking` call in `Connection::new` given its pretty easy to accidentally do the same issue again, and its simpler for our users to not have to think about it than not. Given that, its fewer conversion calls to just accept an `std::net::TcpStream` instead of a tokio one. It doesn't really matter, though, just pretty poor API design on Tokio's part.